tests/support_server/wait_support_server_network.pm

```diff
@@ -0,0 +1,39 @@
+# SUSE's openQA tests
+#
+# Copyright (c) 2016-2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Wait for support server network
+# Maintainer: Petr Cervinka <pcervinka@suse.com>
+
+use base 'opensusebasetest';
+use strict;
+use testapi;
+use serial_terminal 'select_virtio_console';
+
+sub run {
+    select_virtio_console();
+
+    # Check if network is available
+    for (1 .. 5) {
+        eval {
+            # Check support server by ping
+            assert_script_run "ping -c 5 10.0.2.1";
```

Comment:
`assert_script_run` will die if `ping` return a non-zero value, so it may failed on the first loop. I think it's better to use `script_run` instead (be careful, `script_run` can return undefined value).

You could also add a `-w10` option to ping (or less), to be sure that it will failed after a timeout.

Reply:
No, whole code block is designed to don't fail, as you can see in verification: http://10.100.12.105/tests/1176#step/wait_support_server_network/11

Reply:
You're right, I don't exactly how it works but it works :) Maybe it's because the `eval`?

Reply:
Yes, exactly. It is combination of eval and following last unless.